workspace/cannon/app.c

```diff
@@ -96,39 +96,31 @@ void detect_task(intptr_t unused) {
         
         // If the payload length is 0, no block(s) were detected and the loop should be continued
 
-        if(pixycam_response.header.payload_length == 0 || pixycam_response.blocks->signature != signatures) 
+        if(pixycam_response.header.payload_length == 0) 
             continue; // Time: 17
 
+        if(pixycam_response.blocks[0].signature != signatures)
```

Comment:
Split in two because otherwise it would continue to detect something even when there wasn't anything moving.

Also using `[0]` instead of `->`